src/components/fx/hover.js

```diff
@@ -931,10 +931,24 @@ function createHoverText(hoverData, opts, gd) {
             });
         }
 
+        var xLetter = 'x';
+        var yLetter = 'y';
+        var zLetter = 'z';
+
+        if(gd._fullLayout.scene) {
+            xLetter = gd._fullLayout.scene.xaxis.hovertitle || xLetter;
+            yLetter = gd._fullLayout.scene.yaxis.hovertitle || yLetter;
+            zLetter = gd._fullLayout.scene.zaxis.hovertitle || zLetter;
+        }
+        else if(!gd._fullLayout.ternary) {
+            if(gd._fullLayout.xaxis) xLetter = gd._fullLayout.xaxis.hovertitle || xLetter;
+            if(gd._fullLayout.yaxis) yLetter = gd._fullLayout.yaxis.hovertitle || yLetter;
+        }
+
         if(d.zLabel !== undefined) {
-            if(d.xLabel !== undefined) text += 'x: ' + d.xLabel + '<br>';
-            if(d.yLabel !== undefined) text += 'y: ' + d.yLabel + '<br>';
-            text += (text ? 'z: ' : '') + d.zLabel;
+            if(d.xLabel !== undefined) text += xLetter + ': ' + d.xLabel + '<br>';
+            if(d.yLabel !== undefined) text += yLetter + ': ' + d.yLabel + '<br>';
+            text += (text ? zLetter + ': ' : '') + d.zLabel;
```

Comment:
Hmm. So `hovertitle` for cartesian axes only has an effect for trace with a "z" field. That's ok, but should make that clear in the attribute description.